engine/classes/Elgg/Database.php

```diff
@@ -126,6 +127,19 @@ protected function getConnection($type) {
 	}
 
 	/**
+	 * Get an Elgg query builder instance (based on Doctrine DBAL)
+	 *
+	 * This can be passed to most of the public methods that perform queries.
+	 *
+	 * @return QueryBuilder
+	 */
+	public function getQueryBuilder() {
+		// note we supply the readwrite connection, but when we execute in executeQueryBuilder()
+		// we will use the actual connection needed.
```

Comment:
This could backfire if Elgg ever allows different connections to have different DB engines (with different SQL syntaxes). Ways around this I see:

- (preferred) Instead of getting a raw builder, have functions `elgg_build_select()`, `elgg_build_insert()`, et.al., which return the builder set up with the appropriate connection.
- Make user give the connection type (e.g. "read") to get the builder.